File: 7-kyu/least-larger/python/solution.py
def least_larger(a, i):
    differences = {idx: n - a[i] for idx, n in enumerate(a)}
    ret = (-1, float('inf'))
    for k, v in differences.items():
        if 0 < v < ret[1]:
            ret = (k, v)
    return ret[0]

File: 7-kyu/least-larger/javascript/solution.js
const leastLarger = (a, i) => {
  let differences = {};
  for (let idx = 0; idx < a.length; idx++) {
    differences[idx] = a[idx] - a[i];
  }
  let ret = [-1, Number.MAX_SAFE_INTEGER];
  for (const idx in differences) {
    if (0 < differences[idx] && differences[idx] < ret[1]) {
      ret = [idx, differences[idx]];
    }
  }
  return parseInt(ret[0]);
};
